flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/SchemaOperator.java

```diff
@@ -611,14 +614,42 @@ public Object getFieldOrNull(RecordData recordData) {
                 } else if (originalField instanceof Integer) {
                     // INT
                     return ((Integer) originalField).longValue();
+                } else if (originalField instanceof Long) {
+                    // BIGINT
+                    return originalField;
                 } else {
                     return fail(
                             new IllegalArgumentException(
                                     String.format(
                                             "Cannot fit type \"%s\" into a BIGINT column. "
-                                                    + "Currently only TINYINT / SMALLINT / INT can be accepted by a BIGINT column",
+                                                    + "Currently only TINYINT / SMALLINT / INT / LONG can be accepted by a BIGINT column",
+                                            originalField.getClass())));
+                }
+            } else if (destinationType instanceof DecimalType) {
+                DecimalType decimalType = (DecimalType) destinationType;
+                BigDecimal decimalValue;
+                if (originalField instanceof Byte) {
+                    decimalValue = BigDecimal.valueOf(((Byte) originalField).longValue(), 0);
+                } else if (originalField instanceof Short) {
+                    decimalValue = BigDecimal.valueOf(((Short) originalField).longValue(), 0);
+                } else if (originalField instanceof Integer) {
+                    decimalValue = BigDecimal.valueOf(((Integer) originalField).longValue(), 0);
+                } else if (originalField instanceof Long) {
+                    decimalValue = BigDecimal.valueOf((Long) originalField, 0);
+                } else if (originalField instanceof DecimalData) {
+                    decimalValue = ((DecimalData) originalField).toBigDecimal();
+                } else {
+                    return fail(
+                            new IllegalArgumentException(
+                                    String.format(
+                                            "Cannot fit type \"%s\" into a DECIMAL column. "
+                                                    + "Currently only BYTE / SHORT / INT / LONG / DECIMAL can be accepted by a DECIMAL column",
```

Comment:
Can we support merge FLOAT/DOUBLE to DECIMAL like Flink does？ 
https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/types/#casting

Reply:
It's reasonable to have this merging path, I will try to implement it.

Reply:
Unfortunately, it's not possible to determine a "wide enough" Decimal type to store FLOAT / DOUBLE losslessly (since floating point numbers' expression range doesn't have an uplimit -- it could be up to `Infinity`), so maybe it's a not good idea to let this type conversion happen implicitly. Explicit `CAST`ing might be better, I'll open another ticket for this.

Reply:
Makes sense to me.